yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -0,0 +1,350 @@
+		}
+		defer ffdb.Finalize()
+		ffdb.ClearMigrationState(migrationUUID, exportDir)
+	}
+}
+
+func cleanupFallBackDB(msr *metadb.MigrationStatusRecord) {
+	if msr.FallbackEnabled {
+		utils.PrintAndLog("cleaning up fallback db...")
+		fbdb := tgtdb.NewTargetDB(msr.SourceDBAsTargetConf)
+		err := fbdb.Init()
+		if err != nil {
+			utils.ErrExit("end migration: initializing fallback db: %v", err)
+		}
+		defer fbdb.Finalize()
+		fbdb.ClearMigrationState(migrationUUID, exportDir)
+	}
+}
+
+func cleanupExportDir() {
+	utils.PrintAndLog("cleaning up export dir...")
+	subdirs := []string{"schema", "data", "logs", "reports", "temp", "metainfo"}
+	for _, subdir := range subdirs {
+		err := os.RemoveAll(filepath.Join(exportDir, subdir))
+		if err != nil {
+			utils.ErrExit("end migration: removing %s directory: %v", subdir, err)
+		}
+	}
+}
+
+func backupSchemaFilesFn(msr *metadb.MigrationStatusRecord) {
+	if !bool(backupSchemaFiles) {
+		return
+	}
+
+	log.Infof("backing up schema files to %s", backupDir)
+	err := os.MkdirAll(filepath.Join(backupDir, "schema-backup"), 0755)
+	if err != nil {
+		utils.ErrExit("end migration: creating schema backup directory: %v", err)
+	}
+
+	cmd := exec.Command("mv", filepath.Join(exportDir, "schema"), filepath.Join(backupDir, "schema-backup"))
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		utils.ErrExit("end migration: moving schema files: %s: %v", string(output), err)
+	}
+}
+
+func backupDataFilesFn() {
+	if !bool(backupDataFiles) {
+		return
+	}
+
+	log.Infof("backing up sql data files to %s", backupDir)
+	err := os.MkdirAll(filepath.Join(backupDir, "data-backup"), 0755)
+	if err != nil {
+		utils.ErrExit("end migration: creating data backup directory: %v", err)
+	}
+
+	files, err := os.ReadDir(filepath.Join(exportDir, "data"))
+	if err != nil {
+		utils.ErrExit("end migration: reading data directory: %v", err)
+	}
+	for _, file := range files {
+		if file.IsDir() || !strings.HasSuffix(file.Name(), ".sql") {
+			continue
+		}
+
+		err = os.Rename(filepath.Join(exportDir, "data", file.Name()), filepath.Join(backupDir, "data-backup", file.Name()))
+		if err != nil {
+			utils.ErrExit("end migration: moving data files: %v", err)
+		}
+	}
+}
+
+func saveMigrationReportsFn() {
+	if !bool(saveMigrationReports) {
+		return
+	}
+	log.Infof("saving up migration reports to %s", backupDir)
+	// TODO: what if there is report.txt generated from analyze-schema step
+	err := os.Rename(filepath.Join(exportDir, "reports"), filepath.Join(backupDir, "reports"))
+	if err != nil {
+		utils.ErrExit("end migration: moving migration reports: %v", err)
+	}
+
+	utils.PrintAndLog("saving data export reports...")
+	file, err := os.Create(filepath.Join(backupDir, "reports", "export_data_report.txt"))
+	if err != nil {
+		utils.ErrExit("end migration: creating export data status report: %v", err)
+	}
+	originalStdout := os.Stdout
+	os.Stdout = file
+	exportDataStatusCmd.Run(exportDataStatusCmd, nil)
```

Comment:
Consider running the commands in a separate process using `exec.Command("bash", "-c", ..)`.
That way, any surprise changes to global state won't affect the remaining execution of `end migration` command.